### PR TITLE
Autowiki - Generate wiki pages through code

### PR DESCRIPTION
**Pull request by Mothblocks**
## About The Pull Request
Adds a `/datum/autowiki` template which can be derived in order to create wiki pages and queue file uploads. This is then kickstarted by the new tgs target `autowiki` (using the `AUTOWIKI` define) in order to upload these pages.

The pages generated are, in a best case scenario, raw data. This means that wiki editors can decide what the actual theme is without ever having to touch the repository. In the future, MSO will hopefully sandbox the wiki and install [Scribunto](https://www.mediawiki.org/wiki/Extension:Scribunto) to let us separate data and style even more.

These will, when done, upload to templates, such as Template:Autowiki/VendingMachines. The actual pages (in this case "Vending Machines") will include this, and thus can write down their own prose and whatnot without ever having to touch repo.

This will also be run on a daily GitHub action, with some secrets setup to link to the account. Currently this is on a [bot password](https://www.mediawiki.org/wiki/Manual:Bot_passwords) (my forum account will not be leaked in the event of a collapse), but at some point I would like to create a dedicated bot account. 

This PR adds a Techweb and Vending Machine autowiki. You can look at the [Vending Machines one here](https://tgstation13.org/wiki/Template:Autowiki/Content/VendingMachines) and the [Techweb one here](https://tgstation13.org/wiki/index.php?title=Template:Autowiki/Content/Techweb).

I have absolutely no idea what to label this PR (other than note the unit tests I've added). Feel free to add whatever gives GBP 😉 

TODO:
- [x] Daily GitHub action to run the autowiki target (<== This is the big thing)
- [x] Actually make sure the unit tests work (let's see!)
- [x] Aim these at templates like Template:Autowiki/VendingMachines, and not my own user page
- [x] MOTHBLOCKS TODO's

## Why It's Good For The Game

While I would like for the game to make the wiki less necessary, it's undeniably a benefit to the game (even if it is made *unnecessary* in the future). Keeping it up to date is of critical importance.

The more autowikis are added, the less wiki editors have to manually upkeep, meaning the wiki is always up to date and wiki editors can focus their time elsewhere.

### CODE REVIEW

File: code/game/world.dm [hunk 1]
```diff
@@ -260,6 +264,11 @@ GLOBAL_VAR(restart_counter)
 	return
 	#endif
 
+	#ifdef AUTOWIKI
+	qdel(src)
```
Comment (Mothblocks) [inline code/game/world.dm:268]: I don't think I ever actually call Reboot in AUTOWIKI, I qdel the world directly.

**Comment by GoldenAlpharex** [comment]
This looks cool!

**Comment by Bawhoppen** [comment]
This feels like black magic
Do I need to bring the holy water out?

**Comment by Time-Green** [comment]
This is it, the future of SS13 knowledge. Amazing work! You’re making the game better for so many people

**Comment by optimumtact** [comment]
make the vending machine template collapse it's items

**Comment by Time-Green** [comment]
> make the vending machine template collapse it's items

The beauty is that it can all be configured on the wiki 🤩

**Comment by Farquaar** [comment]
The future is here

**Comment by Mothblocks** [comment]
@optimumtact That's for wiki editors to do 😎  (also I think that hurts Ctrl+F? though maybe if only done wrong)

**Comment by optimumtact** [comment]
oh I see it's creating instances of each template

**Comment by Mothblocks** [comment]
yeah isnt that cool

**Pull request merged**